### DISCUSSION TITLE
Transfer Items Between Projects

### DISCUSSION THREAD

**Discussion by dapptain** [community/community#10826] [Projects and Issues]
It would be great if we could transfer an item between projects. 

Down by the "convert to issue" option there would be another option that says "Transfer to Another Project." When clicked, a similar UX to that of the issue creation would show, allowing the user to pick a project to transfer the item to.

**Comment by ADTC** [comment]
Since issues can have multiple projects, you can open the issue, click the gear icon next to Projects and click on the project needed.

**Reply by AlissonRS** [reply]
@ADTC OP is talking about transferring items, not issues, they are not the same. I'd like an option to move items between projects too btw.

**Reply by MikeDrewitt** [reply]
I agree with @AlissonRS. Being able to have a development board and logistics board is better for a more streamlined development process. Being able to easily transfer items between projects would help facilitate this.

I understand that this could largely be accomplished by having multiple columns that go unused, but being able to keep projects completely separated would be better to prevent context switching.

**Reply by zArubaru** [reply]
I find this is one step too many. We should be able to move an issue from within the side panel (which opens when one clicks an issue's name within a project), without having to navigate to the issue's page.

**Comment by riv-dalatorre** [comment]
This would be a very helpful feature, as moving draft pull requests around our project boards would help streamline our planning process. Has there been any movement on this question?

**Comment by junwang-noaa** [comment]
May I ask if there is any progress made to this item? I need to move items from backlog project to a working project, I have to use remove/add approach, which is very inefficient. Thanks

**Comment by jbarnesenginsoft** [comment]
Yes, suffering with this now.  Have accidentally been using a template to store all my items for a long while.  If I click 'use as template' it loses all the items.  So I thought copying the items instead would be the way, but I can't.  If i load from the issues I lose all the additional information in the project table.  For now I will still have to use it as a template project and make the situation worse until I can more the items out of the template project.

**Comment by rebeccabrandt** [comment]
Dealing with this now, as I'm moving all of our issues from a 2023 project to a new 2024 one. When I put the issue into the new project, I lose all of the custom fields assigned and it's a nightmare having to manually reassign everything.

**Reply by devdavedotdev** [reply]
In the same boat😣

**Comment by hartev** [comment]
Same here, moving items between projects would be big help!!

**Comment by bout3fiddy** [comment]
Definitely need this feature! Currently we have to copy paste each ticket item between project boards, and that's a PITA.

**Comment by ericdrosas87** [comment]
My team would definitely make use of this feature - it's much needed!

**Comment by QuintinHumphreys** [comment]
+1

**Comment by urfx** [comment]
I need this feature please.

In the real world we'd walk a card or post it from one board to another. whether that card is an issue, item, task or whatever it needs to move and not hang around getting in the way. Having it (some kind of job card) exist in two places at once would be a recipe for human error. The issue in multiple repos being a case in point.

**Comment by androbi-com** [comment]
We would need this too.

**Comment by andrey-helldar** [comment]
You can easily add tasks to a new project.

Steps:

1. Open the Issues page
2. In the filter, select the project you want to move tasks from
3. In the list, select the tasks to be moved by ticking the checkboxes
4. After that, in the same place where you selected the filter, click again on the menu item “Projects” and select the project where you want to move the tasks. Yes, this is no longer a filter, but action buttons
5. The tasks will be moved to the new project.

<details>
<summary>Show screenshots</summary>

![image](https://github.com/user-attachments/assets/12a0071f-3de6-46e8-9d9c-d2c8b531f03b)
![image](https://github.com/user-attachments/assets/8809514c-3223-4b17-82c9-fcb2060a8171)
![image](https://github.com/user-attachments/assets/0e041ee0-7d00-4578-a73b-aa95ca661fab)
</details>

Nuance: the tasks will be in two projects.

If the goal is to delete the old one, then after these actions the old project is simply deleted.

If the goal is to move some tasks, then these tasks will need to manually disable the project through the menu in the task itself.

The main problem with this approach is that the task list itself will not “disappear”, since they will be in two projects after the changes.

But this can be easily fixed by applying an exclusion filter.

For example, we have two projects - “foo” and “bar”.

We need to move tasks from the “foo” project to the “bar” project and then delete the “foo” project.

Open the Issues page and select the filter for the “foo” project. In the filter field we will see something like ` project:my_repo/1`, where `1` is the project ID. Let's memorize it. Do the same for the project “bar”.

The goal: to quickly get the project identifiers.

For example, it will be ` project:my_repo/1` and ` project:my_repo/2`.

Now we need to enter a condition in the filter field:

```
project:kvedebot/1 -project:kvedebot/2
```

And press “Enter”

After that only those tasks will be displayed in the list, which are in the “foo” project and which are not in the “bar” project.

**Reply by airtonix** [reply]
not talking about issues.

Talking about this 
![image](https://github.com/user-attachments/assets/6247cd05-ba0a-4142-89ca-8044242bc954)

**Comment by rebeccabrandt** [comment]
The above example of moving issues between projects is an okay workaround, however it doesn't take into account the custom fields associated with the original project. For example:

<img width="260" alt="image" src="https://github.com/user-attachments/assets/45e6f485-b498-4edc-aa35-b61aeee2e6e2">

What I have set for priority level, effort, session, theme, and secondary theme will not come over to the new project. You must manually enter all of this info again, and when you're talking about hundreds of tickets... this is a major dealbreaker.

**Reply by andrey-helldar** [reply]
Yeah, it's a big problem if that information needs to be saves.

**Comment by dotysan** [comment]
I didn't have the time to learn their GraphQL and figure out how to preserve all the fields. But here's a simple solution to move between projects using gh cli.
```sh
OWNER=example
SRC_PROJ_NUM=1
DST_PROJ_NUM=2

main() {

    SRC_ITEMS=$(get_proj_items "$SRC_PROJ_NUM")
    # jq . <<<"$SRC_ITEMS"

    local src_title=$(get_proj_title "$SRC_PROJ_NUM")
    local dst_title=$(get_proj_title "$DST_PROJ_NUM")

    local item_id url title item_json
    while read -r item_id url title
    do  echo "Moving item $item_id ($title)..."

        item_json=$(get_item_json "$item_id")
        # jq . <<<"$item_json"

        gh issue edit "$url" --add-project="$dst_title" --add-label="$src_title"

        # TODO: also copy fields (using graphql?)

        # once we've safely copied over all metadata/fields...
        gh issue edit "$url" --remove-project="$src_title"
        echo "Item $item_id moved."

    done < <(get_items)
}

get_proj_items() {
    gh project item-list "$1" --owner="$OWNER" --format=json
}

get_proj_title() {
    gh project view "$1" --owner="$OWNER" --format=json --jq=.title
}

get_items() {
    jq --raw-output '
      .items[] |"\(.id) \(.content.url) \(.title)"
    ' <<<"$SRC_ITEMS"
}

get_item_json() {
    jq ".items[] |select(.id==\"$1\")" <<<"$SRC_ITEMS"
}
```

It cheats and adds a label on the issue with the name of the old project. You must have already created that label in the repo.
